src/foam/dao/jdbc/JDBCPooledDataSource.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.jdbc;
+
+import foam.core.X;
+import org.apache.commons.dbcp2.*;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+import foam.nanos.logger.Logger;
+
+import javax.sql.DataSource;
+import java.sql.DriverManager;
+
+public class JDBCPooledDataSource {
+
+  // Holds a reference to the connection pool ( .getConnection() )
+  protected DataSource dataSource_;
+  protected String poolName_;
+
+  public JDBCPooledDataSource(X x, String poolName) {
+    poolName_ = poolName;
+
+    try {
+      //Add code to get JDBCConnectionSpec properties
+      Object spec = x.get("JDBCConnectionSpec");
+
+      if( spec == null ) {
+        throw new java.sql.SQLException("JDBC connection information not found");
+      }
+
+      if ( !(spec instanceof JDBCConnectionSpec) ) {
+        throw new ClassCastException("Expecting a JDBCConnectionSpec but obtained instead: " + spec.toString());
+      }
+
+      JDBCConnectionSpec jdbcSpec = (JDBCConnectionSpec) spec;
+
+      //jdbc:mysql://localhost/db?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
+      String connectionURI = "jdbc:" + jdbcSpec.getDatabaseServer() + "://" + jdbcSpec.getHostName() +
+        "/" + jdbcSpec.getDatabaseName() + "?useUnicode=true&useJDBCCompliantTimezoneShift=true" +
+        "&useLegacyDatetimeCode=false&serverTimezone=UTC" +
+        "&user=" + jdbcSpec.getUserName() + "&password=" + jdbcSpec.getUserPassword();
+
+      dataSource_ = setupPoolingDataSource(connectionURI, poolName);
+    } catch (ClassNotFoundException e) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error(e);
+    } catch (java.sql.SQLException e) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error(e);
+    }
+
+  }
+
+  public DataSource getDataSource(X x, String poolName) {
```

Comment:
Why does the client need to supply the poolName? Wasn't this instance configured with a specific poolname? If we want to client (DAO) to have a different pool we can just put them in a different sub-context with a different DataSource.

Reply:
No need to provide the poolname, corrected this